commit f9cce1507e30b48e6ea12b9db41e64c6e97aaed3 (refs/changes/11/3211/5)
Author: Orlando Vazquez <ovazquez@gmail.com>
Date:   2018-01-25T16:43:31-08:00 (1 year, 8 months ago)
    
    DOCKER-1140 Handle CNAPI error indicating docker cp'ing is not supported on a stopped container

diff --git a/lib/errors.js b/lib/errors.js
index 7d2d574..e2b4237 100644
--- a/lib/errors.js
+++ b/lib/errors.js
@@ -566,6 +566,21 @@ PathNotDirectoryError.statusCode = 404;
 PathNotDirectoryError.description = 'path was not to a directory';
 
 
+function NotSupportedError(cause, message) {
+    _DockerBaseError.call(this, {
+        restCode: this.constructor.restCode,
+        statusCode: (cause && cause.statusCode) || this.constructor.statusCode,
+        message: message || 'operation not supported',
+        cause: cause
+    });
+}
+util.inherits(NotSupportedError, _DockerBaseError);
+NotSupportedError.prototype.name = 'NotSupportedError';
+NotSupportedError.restCode = 'NotSupported';
+NotSupportedError.statusCode = 409;
+NotSupportedError.description = 'operation not supported';
+
+
 // ---- Volume specific errors
 
 function VolumeSizeNotAvailableError(availableSizes) {
@@ -658,6 +673,11 @@ function cnapiErrorWrap(cause, message, extra) {
         case 'PathNotDirectory':
             return new PathNotDirectoryError();
 
+        case 'DockerCopyStoppedContainerNoPlatformSupport':
+            return new NotSupportedError(cause,
+                '\'docker cp\` on stopped container not'
+                + ' supported on this joyent platform version');
+
         case 'VmNotRunning':
             if (extra && extra.id) {
                 message = 'Container ' + extra.id + ' is not running';
@@ -800,6 +820,7 @@ module.exports = {
     FileNotFoundError: FileNotFoundError,
     NetworkNotFoundError: NetworkNotFoundError,
     PathNotDirectoryError: PathNotDirectoryError,
+    NotSupportedError: NotSupportedError,
 
     VolumeSizeNotAvailableError: VolumeSizeNotAvailableError,
     VolumesNotReachableError: VolumesNotReachableError,
diff --git a/package.json b/package.json
index a809285..f46eb58 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "sdc-docker",
-  "version": "0.4.11",
+  "version": "0.4.12",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
